### PR TITLE
Support multiple build-args

### PR DESCRIPTION
**Pull request by ankurkhemani**
Support `--build-arg` like docker build.
--secret-build-arg for secret args.

can chain multiple build-args:
--build-arg NAME1=value1 --build-arg NAME2=value2

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_params.py [hunk 1]
```diff
@@ -39,6 +39,12 @@ def load_arguments(self, _):
         c.argument('password_name', help='The name of password to regenerate', choices=['password', 'password2'])
         c.argument('username', options_list=['--username', '-u'], help='The username used to log into a container registry')
         c.argument('password', options_list=['--password', '-p'], help='The password used to log into a container registry')
+        c.argument('build_id', help='The unique build identifier.')
+        c.argument('image_name', options_list=['--image', '-t'], help="The image repository and optionally a tag in the 'repository:tag' format.")
+        c.argument('docker_file_path', options_list=['--file', '-f'], help="The relative path of the the docker file to the source code root folder.")        
+        c.argument('build_arg', help='Build argument in a format of <name>=<value>.', action='append')
```
Comment (ankurkhemani) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_params.py:45]: moved common "build" and "build-task" arguments up to "az acr" group.

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/template_build_task_create.json [hunk 2]
```diff
@@ -172,7 +179,8 @@
                 "branch": "[parameters('sourceControlBranch')]",
                 "imageName": "[parameters('imageName')]",
                 "isPushEnabled": "[parameters('isPushEnabled')]",
-                "dockerFilePath": "[parameters('dockerFilePath')]"
+                "dockerFilePath": "[parameters('dockerFilePath')]",
+                "buildArguments": "[parameters('buildArguments')]"
```
Comment (ehotinger) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/template_build_task_create.json:183]: how come secret build args aren't supported here?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py [hunk 3]
```diff
@@ -369,3 +372,16 @@ def _invalid_sku_update():
 
 def _invalid_sku_downgrade():
     raise CLIError("Managed registries could not be downgraded to Classic SKU.")
+
+
+def validate_and_serialize_build_arguments(
```
Comment (ehotinger) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py]: nit: not much serialization going on here, maybe just validate_and_append or something, with a NB to indicate whether or not the original structure will remain immutable or not

**Review by ehotinger — APPROVED** [review]

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/template_build_task_create.json (hunk 2 above)
Comment (ankurkhemani) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/template_build_task_create.json:183]: current RP model combines it with property `isSecret `on `BuildArgument`.

Comment (ehotinger) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/template_build_task_create.json:183]: I see, ty

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py (hunk 3 above)
Comment (ankurkhemani) [reply src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py]: Done.

**Pull request merged**